### ISSUE TITLE
Bug: Malformed data causes really weird behavior

### ISSUE THREAD

**Issue by dplepage** [endless-sky/endless-sky#392]
While playing around with different outfit parameters, I accidentally wrote `"shield generation": 100` instead of `"shield generation" 100`, and that colon somehow caused hostile ships to stop acting entirely. For example, I demanded tribute from a planet, and the planet released its defense fleet, but the ships were all yellow on my radar and all simply drifted outward from the planet - they didn't move, fire, or even turn. Other hostile ships entering the system did the same thing - they hyperjumped in, and then simply drifted across the system and away into the void.

Side note: it might be nice if the data parser complained when it found data it didn't know how to parse, instead of silently failing.

**Comment by frameworld** [comment]
nice cheat ;)

Does neutral fail too? Sounds like a serious failure of the file-protocol-filter / checking-routines.

**Comment by dplepage** [comment]
No, the merchants just went about their business normally. Only the hostile ships froze up.

**Comment by endless-sky** [comment]
In commit 352aaa2 and several commits before it, I've added more detailed error reporting - whenever an unrecognized tag is encountered, plus whenever a non-numeric value is encountered where a number is expected. For example, the error mentioned above where there's an extra ":" would print a message like:

```
Cannot convert value ":" to a number:
  ship "Shield Beetle"
    outfits
      "Pulse Turret" : 4
```

**Comment by endless-sky** [comment]
Included in release 0.8.10.

**Comment by dplepage** [comment]
@endless-sky Thanks for fixing that!
